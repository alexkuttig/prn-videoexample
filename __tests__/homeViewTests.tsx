import React from 'react';
import HomeView from '../src/views/home/Home.view';
import {render, fireEvent, screen} from '@testing-library/react-native';
const genres = require('../assets/data/genres.json');

describe('testing HomeView', () => {
  test('view has not changed', () => {
    const viewJSON = render(
      <HomeView genres={genres} name={'John'} onGenrePress={() => {}} />,
    );
    expect(viewJSON).toMatchSnapshot();
  });

  test('all list items exist', () => {
    render(<HomeView genres={genres} name={'John'} onGenrePress={() => {}} />);
    expect(screen.getByText('Action')).toBeTruthy();
    expect(screen.getByText('Adventure')).toBeTruthy();
    expect(screen.getByText('Animation')).toBeTruthy();
  });
  test('all list items are clickable', () => {
    const mockFn = jest.fn();
    render(<HomeView genres={genres} name={'John'} onGenrePress={mockFn} />);
    fireEvent.press(screen.getByText('Action'));
    fireEvent.press(screen.getByText('Adventure'));
    fireEvent.press(screen.getByText('Animation'));
    expect(mockFn).toBeCalledTimes(3);
  });
  test('click returns valid value', () => {
    const mockFn = jest.fn();
    render(<HomeView genres={genres} name={'John'} onGenrePress={mockFn} />);
    fireEvent.press(screen.getByText('Action'));
    expect(mockFn).toBeCalledWith(genres[0]);
  });
});
