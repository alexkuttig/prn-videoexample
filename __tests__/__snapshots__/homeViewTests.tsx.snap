// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing HomeView view has not changed 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#efefef",
      "padding": 16,
    }
  }
  contentInsetAdjustmentBehavior="automatic"
  style={
    {
      "backgroundColor": "#efefef",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#222222",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      Hello 
      John
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          {
            "backgroundColor": "#e8e8e8",
            "color": "#222222",
            "fontSize": 14,
            "marginBottom": 2,
            "padding": 16,
          }
        }
      >
        Action
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          {
            "backgroundColor": "#e8e8e8",
            "color": "#222222",
            "fontSize": 14,
            "marginBottom": 2,
            "padding": 16,
          }
        }
      >
        Adventure
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          {
            "backgroundColor": "#e8e8e8",
            "color": "#222222",
            "fontSize": 14,
            "marginBottom": 2,
            "padding": 16,
          }
        }
      >
        Animation
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
